modules/Bio/EnsEMBL/Production/Pipeline/Common/AnalysisSetup.pm

```diff
@@ -225,23 +225,18 @@ sub production_updates {
       $self->param($property, $properties{$property});
      }      
    }
+   $sth = $dbh->prepare(
+    'INSERT IGNORE INTO analysis_web_data '.
+      '(analysis_description_id, web_data_id, species_id, db_type, '.
+        'displayable, created_at, modified_at) '.
+    'SELECT '.
+      'ad.analysis_description_id, ad.default_web_data_id, s.species_id, ?, '.
+        'ad.default_displayable, NOW(), NOW() '.
+    'FROM analysis_description ad, species s '.
+    'WHERE ad.logic_name = ? AND ad.is_current = 1 AND s.db_name = ?;'
+   );
   
-   if ($dbc->user eq 'ensrw') {
```

Comment:
@james-monkeyshines: There is already a 'production_lookup' flag, if set to 0 the production database won't get updated (https://github.com/Ensembl/ensembl-production/blob/release/95/modules/Bio/EnsEMBL/Production/Pipeline/Common/AnalysisSetup.pm#L136)
If we need to keep a similar mechanism, we could:
1) Check if the DBC have a password, I think only writable users have a password
2) I could do a eval on the code so if this fail, I would write a warning but the whole pipeline would not fail. 
What do you think?

Reply:
I like the idea of option 1, should be a nice simple change; since we might not need this functionality in the future it's not worth a lot of effort to engineer any thing more robust.

Reply:
Pushed changes for option 1.